nodejs/week3/meal-sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+//...........................................Returns all meals	
+
+router.get("/", async (request, response) => {
+  const {
+    maxPrice,
+    availableReservations,
+    title,
+    createdAfter,
+    limit
+  } = request.query;
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    if (Object.keys(request.query).length === 0) {
+      const resault = await knex("meals").select("*");
```

Comment:
This does work, but when doing a `*` select it is often best to limit the number of results back. Just imagine if there were 1 million+ records, and all of them were pulled from the db every single time this endpoint was called.